airflow/jobs/triggerer_job_runner.py

```diff
@@ -708,3 +710,16 @@ def get_trigger_by_classpath(self, classpath: str) -> type[BaseTrigger]:
         if classpath not in self.trigger_cache:
             self.trigger_cache[classpath] = import_string(classpath)
         return self.trigger_cache[classpath]
+
+    def trigger_row_to_trigger_instance(self, trigger_row: Trigger, trigger_class: type[U]) -> U:
+        """Convert a Trigger row into a Trigger instance."""
+        from airflow.models.crypto import get_fernet
+
+        decrypted_kwargs = {}
+        fernet = get_fernet()
+        for k, v in trigger_row.kwargs.items():
+            if k.startswith("encrypted__"):
+                decrypted_kwargs[k[11:]] = fernet.decrypt(v.encode("utf-8")).decode("utf-8")
```

Comment:
Are you aure this is correct? In here you cur the prefix `encrypted__` off the key but in airflow/models/trigger.py:100 (https://github.com/apache/airflow/pull/36492/files#diff-3d13622d326f0c5cb736bb4fa5b0f3eb4e3d1e86b4b85fb0fb8503a2a8e10a89R100) you don't - should this not be mirroring the logic?

```suggestion
                decrypted_kwargs[k] = fernet.decrypt(v.encode("utf-8")).decode("utf-8")
```

Reply:
The idea is to store the kwarg key in the db with the prefix to know which one we should decrypt when we re-create the trigger, but the trigger __init__ should consume the kwarg without this prefix.